src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -288,17 +288,6 @@ protected virtual Task<Document> PostProcessChangesAsync(Document document, Canc
             return document;
         }
 
-        internal virtual bool PerformFinalApplicabilityCheck => false;
-
-        /// <summary>
-        /// Called by the CodeActions on the UI thread to determine if the CodeAction is still
-        /// applicable and should be presented to the user.  CodeActions can override this if they
-        /// need to do any final checking that must be performed on the UI thread (for example
-        /// accessing and querying the Visual Studio DTE).
-        /// </summary>
-        internal virtual bool IsApplicable(Workspace workspace)
```

Comment:
Oh nice, this was an internal API!

Reply:
Yeah, there is now a possibility we offer a code action that doesn't end up applying changes, but it's a rare edge case at worst and if we need to clean it up we can do so without reintroducing this problematic dependency.